networking/p2p/src/testFixtures/java/tech/pegasys/teku/network/p2p/DiscoveryNetworkFactory.java

```diff
@@ -53,7 +55,8 @@ public DiscoveryNetworkBuilder builder() {
   }
 
   public void stopAll() {
-    networks.forEach(DiscoveryNetwork::stop);
+    SafeFuture.allOf(networks.stream().map(DiscoveryNetwork::stop).toArray(SafeFuture[]::new))
+        .join();
```

Comment:
Probably should use `Waiter.waitFor` here too.

Reply:
Done.